### ISSUE TITLE
Implement Softmax activation

### ISSUE THREAD

**Issue by declspecl** [declspecl/csi4140#5]
Right now we use softmax in CrossEntropy loss

We should separate this out into its own activation function and ensure proper

**Comment by declspecl** [comment]
@antonsakhanovych Ah yeah that's smart, totally didn't think of that. I do think having a separate Softmax activation function is something we definitely want/need - maybe we can design an abstraction that would enable the `Layer` and/or `Loss` to have insight into the specific `Activation` used and optionally perform more optimal calculations? I feel like it could be done using some runtime introspection?
